viewing_party/main.py

```diff
@@ -0,0 +1,292 @@
+                 
+    # return genre_name
+            return genre       
+
+    return None
+
+#----------------------------WAVE 3----------------------------------------------------------------------
+
+def get_unique_watched(user_data):
+    #create a copy of the users watched data as i will remove items from it and dont want it to affect the 
+    #original data
+    users_watched_movies_copy = user_data["watched"].copy()
+    
+    #iterate over the user_data's friends dictionary
+    for friend_dict in user_data["friends"]:
+        
+        #create a new variable to store the friends watched dictionary
+        friend_movies_list = friend_dict["watched"]
+        
+        #iterate over the friends watched movie list to get the elements which is in dict format
+        for friend_movie in friend_movies_list:
+            
+            #if the movie exists in the list call the helper funtions with the given parameters is true then:
+            if (does_movie_exist_in_list(friend_movie["title"], users_watched_movies_copy)):
+                
+                #then remove the movie by calling the helper function
+                remove_duplicate_movie_from_list(friend_movie["title"], users_watched_movies_copy)
+    
+    return users_watched_movies_copy         
+
+
+def get_friends_unique_watched(user_data):
+    #initailize an empty list to store the output
+    friends_unique_watched = []
+    
+    #create a new variable to store the users watched dictionaries
+    users_watched_movies = user_data["watched"]
+    
+    #iterate over the users friends data to ge the friends dictionary 
+    for friend_dict in user_data["friends"]:
+        
+        #create a new variabe to store the friends watched list of dictionaries
+        friend_movies_list = friend_dict["watched"]
+        
+        #iterate over the friends watched list of dictionary (each element is a dictionary)
+        for friend_movie in friend_movies_list:
+            
+            #call the helper function to check if the statement is true: if the movie does not exist is false
+            if (not does_movie_exist_in_list(friend_movie["title"], users_watched_movies)):
+                
+                #then call the helper function with the given parameters to add the friend movie to the list
+                add_movie_to_list(friend_movie, friends_unique_watched)
+    
+    return friends_unique_watched  
+      
+#helper functions
+def does_movie_exist_in_list(movie_title, movie_list):
+    for mov in movie_list:
+        if (mov["title"] == movie_title):
+            return True
+    
+    return False       
+#helper function
+def add_movie_to_list(movie, movie_list):
+    if (not does_movie_exist_in_list(movie["title"], movie_list)):
+        movie_list.append(movie)
+
+#helper function
+def remove_duplicate_movie_from_list(movie_title, movie_list):
+    for i in range(len(movie_list)): 
+        if movie_list[i]['title'] == movie_title: 
```

Comment:
consider a for-each loop:
```suggestion
    for movie in movie_list: 
        if movie['title'] == movie_title: 
```